openapi.go

```diff
@@ -95,6 +95,14 @@ func declareAllTagsFromOperations(s *Server) {
 // Also serves a Swagger UI.
 // To modify its behavior, use the [WithOpenAPIConfig] option.
 func (s *Server) OutputOpenAPISpec() openapi3.T {
+	if s.OpenAPIConfig.Server == nil {
+		s.OpenAPIConfig.Server = &openapi3.Server{
+			URL:         s.proto() + "://" + s.Addr,
```

Comment:
So to sum up, the strong point of this PR is mostly this s.proto am I right?

Nice!

Reply:
No, it's to provide a way to configure a value that we've been setting for our users. This leads into #248. 

If you don't think the configurability is worth it. Please let me just revert the single commit that adds the feature. Thanks